patches/seatd/0001-Add-managarm-specific-changes.patch

```diff
@@ -0,0 +1,200 @@
+-#if defined(__linux__)
++#if defined(__linux__) || defined(__managarm__)
+ 	struct vt_stat st;
+ 	int res = ioctl(fd, VT_GETSTATE, &st);
+ 	close(fd);
+diff --git a/include/drm.h b/include/drm.h
+index 8a7fb10..18d8afa 100644
+--- a/include/drm.h
++++ b/include/drm.h
+@@ -5,7 +5,7 @@ int drm_set_master(int fd);
+ int drm_drop_master(int fd);
+ int path_is_drm(const char *path);
+ 
+-#if defined(__linux__)
++#if defined(__linux__)  || defined(__managarm__)
+ #include <sys/types.h>
+ int dev_is_drm(dev_t device);
+ #endif
+diff --git a/include/evdev.h b/include/evdev.h
+index 6ebd943..1d14c46 100644
+--- a/include/evdev.h
++++ b/include/evdev.h
+@@ -4,7 +4,7 @@
+ int evdev_revoke(int fd);
+ int path_is_evdev(const char *path);
+ 
+-#if defined(__linux__)
++#if defined(__linux__) || defined(__managarm__)
+ #include <sys/types.h>
+ int dev_is_evdev(dev_t device);
+ #endif
+diff --git a/libseat/backend/seatd.c b/libseat/backend/seatd.c
+index 26308d1..fae5adc 100644
+--- a/libseat/backend/seatd.c
++++ b/libseat/backend/seatd.c
+@@ -620,7 +620,7 @@ const struct seat_impl seatd_impl = {
+ 
+ static int set_deathsig(int signal);
+ 
+-#if defined(__linux__)
++#if defined(__linux__) || defined(__managarm__)
+ #include <sys/prctl.h>
+ 
+ static int set_deathsig(int signal) {
+diff --git a/seatd-launch/seatd-launch.c b/seatd-launch/seatd-launch.c
+index 69ca86a..db20d42 100644
+--- a/seatd-launch/seatd-launch.c
++++ b/seatd-launch/seatd-launch.c
+@@ -130,22 +130,22 @@ int main(int argc, char *argv[]) {
+ 
+ 	// Restrict access to the socket to just us
+ 	if (chown(sockpath, uid, gid) == -1) {
+-		perror("Could not chown seatd socket");
+-		goto error_seatd;
++		// perror("Could not chown seatd socket");
++		// goto error_seatd;
+ 	}
+ 	if (chmod(sockpath, 0700) == -1) {
+-		perror("Could not chmod socket");
+-		goto error_seatd;
++		// perror("Could not chmod socket");
++		// goto error_seatd;
+ 	}
+ 
+ 	// Drop privileges
+ 	if (setgid(gid) == -1) {
+-		perror("Could not set gid to drop privileges");
+-		goto error_seatd;
++		// perror("Could not set gid to drop privileges");
++		// goto error_seatd;
+ 	}
+ 	if (setuid(uid) == -1) {
+-		perror("Could not set uid to drop privileges");
+-		goto error_seatd;
++		// perror("Could not set uid to drop privileges");
++		// goto error_seatd;
+ 	}
+ 
+ 	pid_t child = fork();
+diff --git a/seatd/client.c b/seatd/client.c
+index 220c5d3..6d26d83 100644
+--- a/seatd/client.c
++++ b/seatd/client.c
+@@ -24,7 +24,7 @@
+ #include "terminal.h"
+ 
+ static int get_peer(int fd, pid_t *pid, uid_t *uid, gid_t *gid) {
+-#if defined(__linux__)
++#if defined(__linux__) || defined(__managarm__)
+ 	struct ucred cred;
+ 	socklen_t len = sizeof cred;
+ 	if (getsockopt(fd, SOL_SOCKET, SO_PEERCRED, &cred, &len) == -1) {
+@@ -49,6 +49,10 @@ static int get_peer(int fd, pid_t *pid, uid_t *uid, gid_t *gid) {
+ 	*gid = cred.cr_ngroups > 0 ? cred.cr_groups[0] : (gid_t)-1;
+ 	return 0;
+ #else
++	(void) fd;
```

Comment:
I presume this was done to remove warnings, but assuming the `#else` is part of the patched `#if defined(__linux__) || defined(__managarm__)` then this shouldn't be needed I believe?